scripts/ovs/ovs_configurator.py

```diff
@@ -0,0 +1,197 @@
+                tree += "/{}".format(seq)
+
+        validator = getattr(error, "validator")
+        if  validator == 'type':
+            message = '{msg} at {tree}'.format(msg=message, tree=tree)
+        elif validator == 'required':
+            message = "{msg} in config at {tree}. Please check example config for reference.".format(msg=message, tree=tree)
+        raise j.exceptions.RuntimeError(message)
+
+    for key, value in config['network'].items():
+        if 'network' in value:
+            for nodes in config['nodes'].values():
+                _helper(nodes)
+            _helper(config['controller']['hosts'])
+    cmd = 'echo \'{}\' | ssh-keygen -y -f /dev/stdin'.format(config['ssh']['private-key'])
+    _, public_key, _ = j.sal.process.execute(cmd, showout=False)
+    config['ssh']['public-key'] = public_key
+    j.clients.ssh.start_ssh_agent()
+    if j.sal.fs.exists('~/.ssh/id_rsa'):
+        j.clients.ssh.load_ssh_key('~/.ssh/id_rsa')
+    key = config['ssh']['private-key']
+    j.sal.process.execute('echo "%s" | ssh-add /dev/stdin' % key, showout=False)
+    return config
+
+def render(context, loader, filename):
+    data = jinja2.Environment(loader=loader).get_template(filename).render(context)
+    local = j.tools.prefab.local
+    local.core.dir_ensure ('output')
+    with open('output/{}'.format(filename), 'w') as f:
+        f.write(data)
+
+def get_node_disks(nodeip):
+    prefab = j.tools.prefab.getFromSSH(nodeip)
+    rc, out = prefab.core.execute_bash('lsblk -b -d -o  name,size,rota')
+    disks = []
+    if rc == 0:
+        disks_list = out.split('\n')[1:]
+        for disk_obj in disks_list:
+            disk = disk_obj.split()
+            disk_name = disk[0]
+            disk_size = disk[1]
+            disk_type = 'hdd' if disk[2] == '1' else 'ssd'
+            disks.append({'name': disk_name, 'size': float(disk_size)/1024**3, 'type': disk_type})
+    return disks
+
+def enumerate_disks(prefix, disks):
+    enumerated_disks = {}
+    disk_counter = 0
+    ssd_960_counter = 0
+    os_counter = 0
+    nvme_counter = 0
+    for disk in disks:
+        disk_name = '/dev/{}'.format(disk['name'])
+        disk_key = 'do not count'
+        if prefix.startswith('cpu') or disk['type'] == 'hdd':
+            disk_counter += 1
+            disk_key = '{}_disk_{}'.format(prefix, disk_counter)
+        elif prefix.startswith('storage') and disk['type'] == 'ssd':
+            size = disk['size']
+            if disk['name'].startswith('nvme'):
+                nvme_counter += 1
+                disk_key = '{}_disk{}_nvme'.format(prefix, nvme_counter)
+            elif size >= large_ssd and not disk['name'].startswith('nvme'):
+                ssd_960_counter += 1
+                disk_key = '{}_disk{}_ssd_960'.format(prefix, ssd_960_counter)
+            elif size >= large_ssd and not disk['name'].startswith('nvme'):
+                os_counter += 1
+                disk_key = '{}_disk{}_os'.format(prefix, os_counter)
```

Comment:
if a disk has childrne (partitions) then its os disk (disks are wiped when OS is installed)